src/http/http.errors.ts

```diff
@@ -0,0 +1,14 @@
+import { BadRequestError, BaseError, ForbiddenError, InternalServerError, NotFoundError, UnauthorizedError } from '../errors';
+
+export const httpErrorMapping: Record<number, typeof BaseError> = {
+  400: BadRequestError,
+  401: UnauthorizedError,
+  403: ForbiddenError,
+  404: NotFoundError,
+  500: InternalServerError,
+};
+
+const getErrorFromStatusCode = (statusCode: number): typeof BaseError =>
```

Comment:
You don't have to change it. 
But you can make this function use inferred types from the argument as following:

[TS-Playground](http://www.typescriptlang.org/play/#code/MYGwhgzhAEBCkFMCiAnFB7F0DeBfAUKJDPACYBKCAjgK4IQAuqGW2+00AdtALzQCM+AkSjQAqpzA0GAC0wBLAF4JSzTDnZde0AExDC4UQDFMAI3mlSCTmtabufAMz6RMAHLoGJmp1Vp1bBwO0AAsLoYwAJKcDAgokiAAynEAbnG2Ghyk2vwA3PbaAKz6+AgAHgAOmAzQwOicjNAyDAwVtgCyYBUV8pwA5tqBoQAMwwBccGAU1HSMtgA0miHD-BMSUrIKyn4sixzLjhMmKOaW1gtLwyETHl7oPjuYe9CFoxPRsfFgSanp-iiLXD5Qj1Rp9BBMf5GDAAW0SDDADBoEAAwugrNoADwAFWg5VivhgAGsEABPdAAM2gAAoGKSKghKU0Wm1-p1ur0+gBKAB81MYiORaKsE1xAB8uDQYaY4lyJrT6Yyqc1Wh0uj1+lyANrYgC6vB5mgFSNR6IQ0F6zNVbPVnOgAH4rayWOyNX0tcahWboJBoHroBMVc7MK7OVrXsNdcCQQ0aiBtODISxoeg4QiTcKENTliEuflOAgAO7QEDUrkAOk4+Gr5SqKBqVgpUhANUTthTacFpqsuSAA)

Reply:
removed the explicit return types rule from ESLint
changed to infer the return type